pkg/statsd/handler_http_forwarder_v2.go

```diff
@@ -97,25 +102,25 @@ func NewHttpForwarderHandlerV2(logger logrus.FieldLogger, apiEndpoint, network s
 	if maxRequests <= 0 {
 		return nil, fmt.Errorf("max-requests must be positive")
 	}
-	if clientTimeout <= 0 {
-		return nil, fmt.Errorf("client-timeout must be positive")
-	}
 	if maxRequestElapsedTime <= 0 && maxRequestElapsedTime != -1 {
 		return nil, fmt.Errorf("max-request-elapsed-time must be positive")
 	}
 	if flushInterval <= 0 {
 		return nil, fmt.Errorf("flush-interval must be positive")
 	}
 
+	httpClient, err := pool.Get(transport)
+	if err != nil {
+		logger.WithError(err).Error("failed to create http client")
```

Comment:
This isn't always going to be a http client right?

Reply:
It's a `transport.Client`, which has a `Client` member which is an `http.Client`.  Those types won't change, but the usage will change from `transport.Client.Client.Do()` to `transport.Client.PostXyz()` in most places.

Reply:
Wouldn't this error be confusing if left since it could be any of the configured transports in future but we only report that we failed to create a http client?

Reply:
I'm not sure I understand what you mean.  Do you mean it's not obvious which transport has been configured and failed?

Reply:
So if ths could was not to change, but you added another transport. Wouldn't this error be misleading?

Reply:
Gotcha.  Yeah, I'll change it to just "transport" in the next PR.